File: app/src/main/java/com/example/cookbook/repositories/RecipesDataRepository.java
package com.example.cookbook.repositories;

import androidx.lifecycle.LiveData;

import com.example.cookbook.database.dao.RecipeDao;
import com.example.cookbook.models.Recipe;

import java.util.List;

public class RecipesDataRepository {
    private final RecipeDao mRecipeDao;

    public RecipesDataRepository(RecipeDao recipeDao) {
        this.mRecipeDao = recipeDao;
    }

    public LiveData<List<Recipe>> getRecipes() {
        return this.mRecipeDao.getRecipes();
    }
}

File: app/src/main/java/com/example/cookbook/repositories/RecipesDataRepository.kt
package com.example.cookbook.repositories

import androidx.lifecycle.LiveData
import com.example.cookbook.database.dao.RecipeDao
import com.example.cookbook.models.BaseDataRecipe
import com.example.cookbook.models.Recipe

class RecipesDataRepository(private val mRecipeDao: RecipeDao) {
    val recipes: LiveData<List<Recipe>>
        get() = mRecipeDao.getRecipes()

    suspend fun getSpecificRecipe(recipeId: Long): Recipe {
        return mRecipeDao.getSpecificRecipe(recipeId)
    }

    suspend fun insertRecipe(recipe: BaseDataRecipe): Long {
        return mRecipeDao.insertRecipe(recipe)
    }

    suspend fun updateRecipeName(recipeId: Long, name:String){
        mRecipeDao.updateRecipeName(recipeId,name)
    }

    suspend fun updateRecipeUrl(recipeId: Long,recipeUrl:String){
        mRecipeDao.updateRecipeUrl(recipeId,recipeUrl)
    }

}